client/landing/stepper/declarative-flow/new-hosted-site-flow.ts

```diff
@@ -96,19 +98,19 @@ const hosting: Flow = {
 				case 'processing': {
 					// Purchasing Business or Commerce plans will trigger an atomic transfer, so go to stepper flow where we wait for it to complete.
 					const destination = addQueryArgs( '/setup/transferring-hosted-site', {
-						siteId: providedDependencies.siteId,
+						siteId: providedDependencies.siteId || getSignupCompleteSlug(),
```

Comment:
The flow requires us to have `siteId` after checkout. Ideally we should retrieve the id of this stored side when we click back from checkout, but this seems ok for me as well. But we need to double check it.

Reply:
When I get back from checkout and click on a plan, I have the `siteId` with a value, is this necessary?
Yeah, right when we go back from checkout, destination will not have a value for siteId, but it will not be persisted as you changed it to persist only if we have `goToCheckout` true.